src/main/java/org/zalando/nakadi/service/converter/VersionOneConverter.java

```diff
@@ -70,15 +69,37 @@ public NakadiCursor convert(final String eventTypeStr, final Cursor cursor)
         } catch (final NumberFormatException ex) {
             throw new InvalidCursorException(CursorError.INVALID_OFFSET);
         }
-        final List<Timeline> timelines = eventTypeCache.getTimelinesOrdered(eventTypeStr);
-        final Timeline timeline = timelines.stream()
-                .filter(t -> t.getOrder() == order)
-                .findAny()
-                .orElseThrow(() -> new InvalidCursorException(CursorError.UNAVAILABLE));
+        return findCorrectTimelinedCursor(eventTypeStr, order, cursor.getPartition(), offsetStr);
+    }
+
+    private NakadiCursor findCorrectTimelinedCursor(
+            final String eventType, final int order, final String partition, final String offset)
+            throws InternalNakadiException, NoSuchEventTypeException, InvalidCursorException {
+        final List<Timeline> timelines = eventTypeCache.getTimelinesOrdered(eventType);
+        final Iterator<Timeline> timelineIterator = timelines.iterator();
+        Timeline timeline = null;
+        while (timelineIterator.hasNext()) {
+            final Timeline t = timelineIterator.next();
+            if (t.getOrder() == order) {
+                timeline = t;
+                break;
+            }
+        }
+        if (null == timeline) {
+            throw new InvalidCursorException(CursorError.UNAVAILABLE);
+        }
+        String offsetReplacement = offset;
+        while (StaticStorageWorkerFactory.get(timeline.getStorage())
+                .isLastOffsetForPartition(timeline, partition, offsetReplacement)) {
+            // Will not check this call, because latest offset is not set for last timeline
+            timeline = timelineIterator.next();
+            offsetReplacement = StaticStorageWorkerFactory.get(timeline.getStorage())
```

Comment:
I didn't understand this corner case. Could you please give an example? Like, if the we have:

`[1 2 3] -> []`

the cursor transformation from `001-0001-000000000000000003` to `order: 1, offset: -1`?

Reply:
I rewrote tests to clarify all the patterns used https://github.com/zalando/nakadi/pull/797/commits/62dc085236c93ea7700017aaec23b51d02310a4c